xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-focusedplaceholder/plugin.js

```
@@ -0,0 +1,268 @@
+                    h2: "xwiki-focusedplaceholder.placeholder.h2", // Heading 2
+                    h3: "xwiki-focusedplaceholder.placeholder.h3", // Heading 3
+                    h4: "xwiki-focusedplaceholder.placeholder.h4", // Heading 4
+                    h5: "xwiki-focusedplaceholder.placeholder.h5", // Heading 5
+                    h6: "xwiki-focusedplaceholder.placeholder.h6", // Heading 6
+                    main: "xwiki-focusedplaceholder.placeholder.main", // Main content
+                    nav: "xwiki-focusedplaceholder.placeholder.nav", // Navigation
+                    section: "xwiki-focusedplaceholder.placeholder.section", // Section
+                    blockquote: "xwiki-focusedplaceholder.placeholder.blockquote", // Quote
+                    dl: "xwiki-focusedplaceholder.placeholder.dl", // Definition list
+                    dt: "xwiki-focusedplaceholder.placeholder.dt", // Definition entry
+                    dd: "xwiki-focusedplaceholder.placeholder.dd", // Description
+                    figcaption: "xwiki-focusedplaceholder.placeholder.figcaption", // Caption
+                    figure: "xwiki-focusedplaceholder.placeholder.figure", // Figure
+                    ol: "xwiki-focusedplaceholder.placeholder.ol", // Ordered list
+                    ul: "xwiki-focusedplaceholder.placeholder.ul", // Unordered list
+                    menu: "xwiki-focusedplaceholder.placeholder.menu", // Menu (unordered list)
+                    li: "xwiki-focusedplaceholder.placeholder.li", // List item
+                    p: "xwiki-focusedplaceholder.placeholder.p", // Paragraph
+                    pre: "xwiki-focusedplaceholder.placeholder.pre", // Preformated
+                    caption: "xwiki-focusedplaceholder.placeholder.caption", // Title
+                    th: "xwiki-focusedplaceholder.placeholder.th", // Header
+                    td: "xwiki-focusedplaceholder.placeholder.td", // Cell
```

I wouldn't make this default configuration explicit. Instead, given a ``tagName`` I would:
```
var placeholder = editor.config.focusedplaceholder.placeholder[tagName];
if (typeof placeholder !== 'string') {
  placeholder = editor.localization.get('xwiki-focusedplaceholder.placeholder.' + tagName);
}
```

---

I understand the idea, but I'm not sure how to properly handle non existing translation keys.
The localization.get call would fail silently.

---

Why would the translations be missing? That would be a bug no? If you really want to bulletproof the code then you can check if the output of ``localization.get`` is the translation key, which means the translation is missing.